refinery/tool_manager/views.py

```diff
@@ -80,69 +63,32 @@ def get_queryset(self):
-
-    def list(self, request, *args, **kwargs):
-        return super(ToolDefinitionsViewSet, self).list(
-            request,
-            data_set_uuid_lookup_name="dataSetUuid"
-        )
+        else:
+            return []  # get_queryset should return an iterable
 
 
 class ToolsViewSet(ToolManagerViewSetBase):
-    """API endpoint that allows for Tools to be fetched and launched"""
-    queryset = Tool.objects.all()
+    """ViewSet that allows for Tools to be fetched, launched and relaunched"""
     serializer_class = ToolSerializer
     lookup_field = 'uuid'
     http_method_names = ['get', 'post']
 
-    def list(self, request, *args, **kwargs):
-        self.data_set_uuid = request.query_params.get('data_set_uuid')
-        if self.data_set_uuid:
-            return super(ToolsViewSet, self).list(
-                request,
-                data_set_uuid_lookup_name="data_set_uuid"
-            )
-        return super(ToolManagerViewSetBase, self).list(request)
-
     def get_queryset(self):
-        """
-        This view returns a list of all the Tools that the currently user has
-        at least read_meta permissions on.
-        """
-        # returns user's owned tools
-        if not self.data_set_uuid:
-            vis_tools = [v for v in VisualizationTool.objects.all()
-                         if v.get_owner() == self.request.user]
-
-            workflow_tools = [w for w in WorkflowTool.objects.all()
-                              if w.get_owner() == self.request.user]
-
-            tool_type = self.request.query_params.get("tool_type")
-            if tool_type == ToolDefinition.VISUALIZATION.lower():
-                return vis_tools
-            elif tool_type == ToolDefinition.WORKFLOW.lower():
-                return workflow_tools
-            else:
-                user_tools = vis_tools
-                user_tools.extend(workflow_tools)
-                return user_tools
-
-        elif self.request.user.has_perm('core.read_meta_dataset',
-                                        self.data_set):
-            tool_type = self.request.query_params.get("tool_type")
-
-            if not tool_type:
-                return self.user_tools
-
-            tool_types_to_tools = {
-                ToolDefinition.VISUALIZATION.lower(): self.visualization_tools,
-                ToolDefinition.WORKFLOW.lower(): self.workflow_tools
-            }
-            # get_queryset should return an iterable
-            return tool_types_to_tools.get(tool_type.lower()) or []
-
-        return Response("User is not authorized to view visualizations.",
-                        status=status.HTTP_401_UNAUTHORIZED)
+        visualization_tools = list(
+            VisualizationTool.objects.filter(dataset=self.data_set)
+        )
+        workflow_tools = list(
+            WorkflowTool.objects.filter(dataset=self.data_set)
+        )
+        tool_type = self.request.query_params.get("tool_type")
+        if tool_type is None:
```

Comment:
@scottx611x Any user can hit this endpoint and grab analyses or visualizations with a dataSetUuid. Thoughts?

Reply:
Ah, this is just the `get_queryset` method. The permission logic happens in the respective `list()` methods